orchestrator-salt/src/main/resources/salt/salt/post-recipes/scripts/createuserhome.sh

```diff
@@ -71,25 +71,47 @@ PATH_PREFIX="/user"
 
 export KRB5CCNAME=/tmp/krb5cc_cloudbreak_$EUID
 
+
+klist -s
+if [[ $? -ne 0 ]]; then
+  kinit_as_hdfs
+fi
+
 WEBHDFS_HTTP_POLICY=$(hdfs getconf -confkey dfs.http.policy)
 
-if [ "$WEBHDFS_HTTP_POLICY" == "HTTP_ONLY" ]; then
-  NAMENODE_HTTP_ADDRESS=$(hdfs getconf -confkey dfs.namenode.http-address)
-  WEBHDFS_URL=http://$NAMENODE_HTTP_ADDRESS/webhdfs/v1
+hdfs haadmin -getAllServiceState
+if [[ $? -ne 0 ]]; then
+  echo "Single node configuration"
+  if [ "$WEBHDFS_HTTP_POLICY" == "HTTP_ONLY" ]; then
+    NAMENODE_HTTP_ADDRESS=$(hdfs getconf -confkey dfs.namenode.http-address)
+    WEBHDFS_URL=http://$NAMENODE_HTTP_ADDRESS/webhdfs/v1
+  else
+    NAMENODE_HTTP_ADDRESS=$(hdfs getconf -confkey dfs.namenode.https-address)
+    WEBHDFS_URL=https://$NAMENODE_HTTP_ADDRESS/webhdfs/v1
+  fi
 else
-  NAMENODE_HTTP_ADDRESS=$(hdfs getconf -confkey dfs.namenode.https-address)
-  WEBHDFS_URL=https://$NAMENODE_HTTP_ADDRESS/webhdfs/v1
+  echo "HA configuration"
+  NN_HOST=$(hdfs haadmin -getAllServiceState | grep active | cut -d':' -f 1)
+  echo "Active NAMENODE host: $NN_HOST"
+  if [ "$NN_HOST" != "$(hostname -f)" ]; then
+    echo "This is not the active host. Script will run on the active one"
+    exit 0
+  fi
+  if [ "$WEBHDFS_HTTP_POLICY" == "HTTP_ONLY" ]; then
+    NAMENODE_PORT=$(hdfs getconf -confkey dfs.namenode.http-address | cut -d':' -f 2)
+    echo "NN port: $NAMENODE_PORT"
+    WEBHDFS_URL=http://$NN_HOST$NAMENODE_PORT/webhdfs/v1
```

Comment:
I'm guessing you need the colon here between host and port if I'm reading this correctly?

Reply:
yep, fixed. good catch!